doc/userguide/rules/file-keywords.rst

```diff
@@ -5,21 +5,25 @@ Suricata comes with several rule keywords to match on various file
 properties. They depend on properly configured
 :doc:`../file-extraction/file-extraction`.
 
-filename
---------
+file.name
+---------
 
 Matches on the file name.
 
 Syntax::
 
-  filename:<string>;
+  file.name:<string>;
 
 Example::
 
-  filename:"secret";
+  file.name:"secret";
 
 ``file.name`` supports multiple buffer matching, see :doc:`multi-buffer-matching`.
 
+**Note** The ``filename`` keyword is still supported but the
```

Comment:
this is no longer true:

`filename:"secret";` is now equivalent to `file.name; content:"secret"; nocase;`

Similarly, fileext:
`fileext:pdf;` is equivalent to `file.name; content:".pdf"; nocase; endswith;`. Note the dot that is prepended to `pdf`.

This is handled at parsing, so at runtime this is identical in behavior and performance.

btw we're looking at backporting this optimization to 6: https://redmine.openinfosecfoundation.org/issues/6203